extensions/sql-database-projects/src/sqldbproj.d.ts

```diff
@@ -37,21 +37,29 @@ declare module 'sqldbproj' {
 		/**
 		 * Adds the list of sql files and directories to the project, and saves the project file
 		 * @param list list of files and folder Uris. Files and folders must already exist. No files or folders will be added if any do not exist.
+		 * @param ignoreDuplicates Flag that indicates whether duplicate entries should be ignored.
 		 */
-		addToProject(list: vscode.Uri[]): Promise<void>;
+		addToProject(list: vscode.Uri[], ignoreDuplicates?: boolean): Promise<void>;
```

Comment:
I did consider using the inverted name (i.e. `throwIfDuplicate`), but decided against it, because to preserve existing default behavior it would have to be set to `true`. Given that this code base prefers `T | undefined` instead of explicit defaults and `true` is rarely a default for boolean type, I decided to use the non-inverted name that aligns with the new feature that is being added, rather than original default behavior. This also exposes lack of default values for interfaces in TypeScript with all consequences, i.e. one cannot do `throwIfDuplicate: boolean = true` for an interface method, so interface will be `throwIfDuplicate?: boolean`, but class will be `throwIfDuplicate: boolean = true`.
After all, I don't see how the inverted name solves the suggested "ambiguity", as it still does not indicate what is the alternative (to throwing, in that case) - skip or add. I can suggest `doNotThrowOnDuplicate`, which will address the above and presumably eliminate the confusion around failing behavior, which, being the original default, I have never thought was a confusion at first place.

I'm not really sure if this deserves an enumeration, given that there are just two values at the moment. This can be refactored later if additional duplicate handling strategies are possible (which they are). Keep in mind that enumeration also has the problem with obscure default, where a boolean flag with `false` default is less so.

Reply:
I agree that it an enum would be overkill here, but it would also leave the least open to different interpretations.  `doNotThrowOnDuplicate` is fine as well.

Reply:
Renamed the argument to `doNotThrowOnDuplicate`. Hopefully nobody complains that it's now unclear what is the behavior when we do not throw. 🤦